lib/THC/THCTensorMathMagma.cu

```diff
@@ -0,0 +1,472 @@
+#include "THCTensorMath.h"
+#include "THCTensorCopy.h"
+
+#define NoMagma(name) "No CUDA implementation of '" #name "'. Install MAGMA and rebuild cutorch (http://icl.cs.utk.edu/magma/)"
+
+
+#ifdef USE_MAGMA
+static inline float* th_magma_smalloc_pinned(size_t n)
+{
+  float* ptr;
+  if (MAGMA_SUCCESS != magma_smalloc_pinned(&ptr, n))
+    THError("$ Torch: not enough memory: you tried to allocate %dGB. Buy new RAM!", n/268435456);
+  return ptr;
+}
+
+static inline int* th_magma_imalloc_pinned(size_t n)
+{
+  int* ptr;
+  if (MAGMA_SUCCESS != magma_imalloc_pinned(&ptr, n))
+    THError("$ Torch: not enough memory: you tried to allocate %dGB. Buy new RAM!", n/268435456);
+  return ptr;
+}
+
+static void THCudaTensor_copyArray1d(THCState *state, THCudaTensor *self, float *src, int k)
+{
+  long size[1] = { k };
+  long stride[1] = { 1 };
+  THCudaTensor_rawResize(state, self, 1, size, stride);
+  size_t len = k * sizeof(float);
+  THCudaCheck(cudaMemcpy(self->storage->data + self->storageOffset, src, len, cudaMemcpyHostToDevice));
```

Comment:
I believe it does ensure that self is contiguous since it ensures the stride is 1.